feat: 프로필 페이지 닉네임 입력 부분 수정

yeni023 opened the pull request:
## 🔗 반영 브랜치
- closed #189 
- `feature/profile-enter-key-submit → dev`

## 📝 작업 내용
Profile 페이지 닉네임 입력창에서 Enter 키로도 닉네임 변경이 가능하도록 기능 추가

## 💬 리뷰 요구사항(선택)
.

yeni023 committed:
feat: allow nickname submission with Enter key

yeni023 committed:
fix: prevent default behavior on Enter key when submitting nickname

yeni023 committed:
fix: prevent duplicate nickname update requests in Profile page

yeni023 committed:
feat: autofocus nickname input in profile edit modal

hakilee commented on src/pages/ProfilePage/components/nickname-edit-dialog/index.tsx:
```diff
@@ -15,42 +15,66 @@ import * as styles from './styles.css';
         <DialogClose />
 
         <DialogTitle className={styles.title}>Change nickname</DialogTitle>
         <input
```
인풋에 직접적으로 onKeyDown 을 넣지 말고, input과 button을 form으로 감싼 후에, form에 onSubmit 을 넣는 것은 어떨까요?

현재 방법을 그대로 사용해도 괜찮을 것 같지만, 시맨틱 적으로 조금 더 명확할 것 같아요!

hakilee reviewed (commented):
고생하셨습니다~!!

yeni023 committed:
refactor: 닉네임 수정 시 중복 요청 방지 및 포커싱 방식 개선

- useEffect 대신 onOpenAutoFocus로 포커싱 처리

- isSubmitting 상태 제거하고 react-query의 isPending 사용

- input에 직접 이벤트 등록 대신 form onSubmit 구조로 변경하여 시맨틱 구조 개선

hakilee commented on src/pages/ProfilePage/components/nickname-edit-dialog/index.tsx:
```diff
@@ -38,21 +41,35 @@ export const NicknameEditDialog = ({ children }: PropsWithChildren) => {
+          e.preventDefault();
+          inputRef.current?.focus();
+        }}
+      >
```
```suggestion
      <DialogContent
        className={styles.content}
        onOpenAutoFocus={(e) => {
          e.preventDefault();
          inputRef.current?.focus();
        }}
        aria-describedby={undefined}
      >
```

hakilee reviewed (commented):
Dialog에 Description을 사용하지 않는 경우, `undefined`를 명시적으로 작성해주는 것이 좋을 것 같아요. 작성 하지 않으면 콘솔에 워링이 떠 있을 거에요.

HJ0311 reviewed (dismissed):
수고하셨습니다!

yeni023 committed:
fix: add undefined to Dialog to avoid warning
